src/hotspot/share/opto/memnode.cpp

```diff
@@ -2790,22 +2790,30 @@ class MergePrimitiveStores : public StackObj {
 private:
   PhaseGVN* const _phase;
   StoreNode* const _store;
+  // State machine with initial state Unknown
+  // Allowed transitions:
+  //   Unknown  -> Forward
+  //   Unknown  -> Backward
+  //   Forward  -> Forward
+  //   Backward -> Backward
+  enum ValueOrder { Unknown, Forward, Backward };
```

Comment:
can we update it to: 
```suggestion
  enum ValueOrder : uint8_t { Unknown, Forward, Backward };
```